proofs_test.go

```diff
@@ -1,250 +1,247 @@
 package smt
```

Comment:
One general remark, this could probably benefit from using https://github.com/golang/go/wiki/TableDrivenTests